idl/testing/run_tests.py

```diff
@@ -58,8 +58,227 @@
+    '--wildcard',
+    default='SUBSCRIPT(GETNCI("\\\\CMOD::TOP.***:CPLD_START", "FULLPATH"),0)',
+    help='An expression to evaluate and compare against --wildcard-value'
+)
+
+parser.add_argument(
+    '--wildcard-value',
+    default='\CMOD::TOP.DNB.MIT_CXRS:CPLD_START',
+    help='The value of evaluating the --wildcard expression, ignoring leading/trailing whitespace'
+)
+
+# The parser is converting "\\" into "\", so to get desired path of "\\CMOD::" must use "\\\\"
+parser.add_argument(
+    '--getnci',
+    default=' SUBSCRIPT(GETNCI("\\\\CMOD::TOP.DNB.MIT_CXRS:CPLD_START", "USAGE"),0) ',
+    help='An expression to evaluate and compare against --getnci-value'
+)
+
+parser.add_argument(
+    '--getnci-value',
+    default='5',
+    help='The value of evaluating the --getnci expression, ignoring leading/trailing whitespace'
+)
+
+parser.add_argument(
+    '--rank',
+    default='RANK(\\IP)',
+    help='An expression to evaluate and compare against --rank-value'
+)
+
+parser.add_argument(
+    '--rank-value',
+    default='1',
+    help='The value of evaluating the --rank expression, ignoring leading/trailing whitespace'
+)
+
+parser.add_argument(
+    '--ndims',
+    default='NDIMS(\\IP)',
+    help='An expression to evaluate and compare against --ndims-value'
+)
+
+parser.add_argument(
+    '--ndims-value',
+    default='1',
+    help='The value of evaluating the --ndims expression, ignoring leading/trailing whitespace'
+)
+
+parser.add_argument(
+    '--dim-of',
+    default='DIM_OF(\\CMOD::TOP.ADMIN.FAST_WINDOW.XTOMO:OPTIONS, 0)',
+    help='An expression to evaluate and compare against --dim-of-value'
+)
+
+parser.add_argument(
+    '--dim-of-value',
+    default='1           1           1           1',
+    help='The value of evaluating the --dim-of expression, ignoring leading/trailing whitespace'
+)
+
+parser.add_argument(
+    '--units-of',
+    default='UNITS_OF(\\CMOD::TOP.MHD.XTOMO.SIGNALS.ARRAY_1:CHORD_01)',
+    help='An expression to evaluate and compare against --units-of-value'
+)
+
+parser.add_argument(
+    '--units-of-value',
+    default='watts',
+    help='The value of evaluating the --units-of expression, ignoring leading/trailing whitespace'
+)
+
+parser.add_argument(
+    '--write-tree',
+    default='idl_tests',
+    help='Name of tree to create for exercising write functions of the API'
+)
+
+parser.add_argument(
+    '--write-shot',
+    default=100,
+    help='Shot number of tree to create for exercising write functions of the API'
+)
+
 args = parser.parse_args()
 
+
+#---------------------------------------------------------------------------
+# Each IDL write test should start with a clean tree.
+# Write tests use a local tree, but eventually will be upgraded to use mdsip.
+# TODO: The default_tree_path is temporary; will change when switch to mdsip.
+#
+def build_write_tree(tree, shot):
+    import os
+    import MDSplus as mds
+
+    test_dir = os.getenv('MDSPLUS_DIR') + '/idl/testing/'
```

Comment:
We should instead use a `tempfile.TemporaryDirectory`, but be sure to call that once at the top, not once for each test

Reply:
That is a great suggestion!

Reply:
The `/idl/testing` directory is where we already create the transient `test.pro` files that `run_tests.py` executes.   Should those `test.pro` files also be created in a `tempfile.TemporaryDirectory`?

The local tree used for testing the write features of the IDL API is only ~3 KB in size.   Which is why I placed it in the same `idl/testing` directory as the transient `test.pro` files.

Regardless of which directory is used, I favor having the transient `test.pro` files and the dinky local "write" tree in the same directory.  But am open to using two different directories if that makes Jenkins more robust.

Also note that after we upgrade the infrastructure to do mdsip testing and cross-version testing, instead of a local tree we instead need to create the dinky "write" tree on the remote server. 

Regardless, I am now experimenting with `temple.TemporaryDirectory`.   But will hold off on making that commit until we decide whether to move the transient `test.pro` files too.

Reply:
Actually, yeah we should probably put the `.pro` there, it'll keep a truly temporary file out of the working directories